DICOM/src/Microsoft.Health.DeID.SharedLib/EncryptFunction.cs

```diff
@@ -0,0 +1,149 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib.Exceptions;
+
+namespace Microsoft.Health.Dicom.DeID.SharedLib
+{
+    public class EncryptFunction
+    {
+        // AES Initialization Vector length is 16 bytes
+        private const int AesIvSize = 16;
+        private readonly byte[] _aesKey;
+
+        public EncryptFunction(EncryptSetting encryptionSetting = null)
+        {
+            encryptionSetting ??= new EncryptSetting();
+            encryptionSetting.Validate();
+
+            _aesKey = Encoding.UTF8.GetBytes(encryptionSetting.EncryptKey);
+        }
+
+        public byte[] Encrypt(string plainText, Encoding encoding = null)
+        {
+            EnsureArg.IsNotNull(plainText, nameof(plainText));
+
+            if (plainText == string.Empty)
+            {
+                return new byte[] { };
+            }
+
+            encoding ??= Encoding.UTF8;
+            return Encrypt(encoding.GetBytes(plainText));
+        }
+
+        public byte[] Encrypt(Stream plainStream)
+        {
+            EnsureArg.IsNotNull(plainStream, nameof(plainStream));
+
+            return Encrypt(StreamToByte(plainStream));
+        }
+
+        public byte[] Encrypt(byte[] plainBytes)
+        {
+            EnsureArg.IsNotNull(plainBytes, nameof(plainBytes));
+
+            /* Create AES encryptor:
+             * Mode: CBC
+             * Block size: 16 bytes
+             * Acceptable key sizes： [128, 192, 256]
+             */
+            try
+            {
+                using Aes aes = Aes.Create();
+                byte[] iv = aes.IV;
+                aes.Key = _aesKey;
+                ICryptoTransform encryptor = aes.CreateEncryptor(aes.Key, aes.IV);
+
+                var encryptedBytes = encryptor.TransformFinalBlock(plainBytes, 0, plainBytes.Length);
```

Comment:
🔴 [Insist]

Using TransformFinalBlock is only encrypting the final block. Please refer to [AesCryptoServiceProvider Class](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.aescryptoserviceprovider?view=net-5.0) for more information.